### ISSUE TITLE
Consider adding __cdk_name

### ISSUE THREAD

**Issue by lastmjs** [demergent-labs/azle#693]
It would be awesome to allow canister indexing tools to discover which CDK canisters were written in. This should be easily identifiable by doing something like the following in the Candid files:

```
type CDK = { azle };
```

Update: We should just generate this into lib.rs
```rust
#[ic_cdk_macros::query]
fn __cdk_name() -> String {
"azle".to_string()
}
```

**Comment by lastmjs** [comment]
I think it's best to move forward with what is being discussed here: https://github.com/dfinity/motoko/issues/3462#issuecomment-1258599165

**Comment by lastmjs** [comment]
Do it like this: https://github.com/Psychedelic/ic-kit/commit/082ea13d6cfbd2d7b945da45ae4d9530960be1df#diff-5dd55cfa837d6176224433e3a1644424adaaa010c65d7d3bdbb95ec9abff629d

**Comment by dansteren** [comment]
Closely related to #692 and will be resolved with #822
